src/main/java/com/google/codeu/servlets/StatsPageServlet.java

```diff
@@ -0,0 +1,41 @@
+package com.google.codeu.servlets;
+
+import java.io.IOException;
+
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.codeu.data.Datastore;
+import com.google.gson.JsonObject;
+/**
+ * Handles fetching site statistics. 
+ * 
+ * This uses the init() function that creates and store a Datastore instance,
+ * then doGet() calls the function getTotalMessageCount() in the Datastore.java
+ * and create a JSON that countains the message count
+ */
+public class StatsPageServlet extends HttpServlet{
+  private Datastore datastore;
+
+  @Override
+  public void init() {
```

Comment:
Is there a reason that you do this initialization here rather than statically or via a constructor? (I'm not saying that you need to do it differently, I'd just like to better understand this particular design decision).

Reply:
I believe the intention was doing the initialization every time that the stats servlet is created. I actually dont know why, it could be later on that it becomes pertinent when we start implementing more things that are user focused.

Reply:
I suspect that the example code might do it this way for testing reasons, though -- if that is the case -- it would probably make sense to change this to:

```
public void init() {
   super.init();
   datastore = createDatastore();
}

/*
 * Instantiates a connection to a datastore. In tests, override this with a method that
 * returns a fake/mock datastore object.
 */
protected Datastore createDatastore() {
   return new Datastore();
}
```

As-is, this code does not provide a way to replace the `datastore` with a fake/mock version. Given that, it doesn't make a whole lot of sense to instantiate a separate `Datastore` per servlet rather than having simply:

```
   private static Datastore datastore = new Datastore();
```

That being said, I would encourage going with the former rather than the latter (because testing is important and we should write code in a manner that lends itself to being easily tested).